src/data/buffer/parser/basic.lean

```diff
@@ -65,6 +63,14 @@ by cases p cb n; simp
 
 variables {p q cb n n' msgs msg}
 
+lemma mono.le (hp : p.mono) (cb : char_buffer) (n : ℕ) : n ≤ (p cb n).pos := mono.le' cb n
```

Comment:
Why bother having `mono.le'` and `mono.le` as the same thing?

Reply:
Without having `mono.le`, the proofs below of
```
lemma mono.of_done [hp : p.mono] (h : p cb n = done n' a) : n ≤ n' :=
by simpa [h] using mono.le hp cb n
```
won't work nicely.

Reply:
Oh, so the difference is the binding of the `hp` argument. A comment might be good.

Reply:
Arguably, `p` should be explicit here and `(hp : p.mono)` should be `[p.mono]`. I think this helps about as much as it hurts in the rest of the file. The signature that `class` generates automatically for `le'` is the least useful of all the combinations...